package/kernel-osmc/initramfs-src/init

```diff
@@ -83,8 +83,37 @@ for option in $(/bin/busybox cat /proc/cmdline); do
 	    ;;
 	  root=*)
 		OPTION_ROOT=$(echo "${option#*=}" | /bin/busybox sed 's/"//g')
+		ROOT_STR=$OPTION_ROOT
 		PARTUUID=$(echo "$OPTION_ROOT" | /bin/busybox grep "^PARTUUID=" | /bin/busybox sed 's/^PARTUUID=//g')
-		if [ -n "$PARTUUID" ]; then OPTION_ROOT=$(/bin/busybox readlink -f /dev/disk/by-partuuid/${PARTUUID}); fi
+		UUID=$(echo "$OPTION_ROOT" | /bin/busybox grep "^UUID=" | /bin/busybox sed 's/^UUID=//g')
+		if ! [ -z "$UUID$PARTUUID" ]; then
+			# it takes a while for USB partitions to show up
+			count=12
+			OPTION_ROOT=
+			while [ $count -ge 0 -a -z "$OPTION_ROOT" ]; do
+				echo $count seconds
+				echo $(/bin/busybox blkid)
+				if [ -n "$PARTUUID" ]; then OPTION_ROOT=$(/bin/busybox findfs "PARTUUID=${PARTUUID}"); fi
+				if [ -n "$UUID" ]; then OPTION_ROOT=$(/bin/busybox findfs "UUID=${UUID}"); fi
+				if ! [ -z "$OPTION_ROOT" ]; then break; fi
+				count=$((count - 1))
+				/bin/busybox sleep 1
+			done
+		fi
+		# fallbacks if dev or UUIDs not found
+		if [ -z "$OPTION_ROOT" -o -z "$PARTUUID$UUID" ]; then
```

Comment:
I don't think this is appropriate. I think we want our current, existing fallback as defined in init.d. It breaks the encapsulation approach for initramfs.

Reply:
I'll check that.